rest_framework/validators.py

```diff
@@ -188,7 +188,10 @@ def __call__(self, attrs, serializer):
                 if attrs[field_name] != getattr(serializer.instance, field_name)
             ]
 
-        condition_kwargs = {source: attrs[source] for source in self.condition_fields}
+        condition_kwargs = {
+            serializer.fields[field_name].source: attrs[serializer.fields[field_name].source]
+            for field_name in self.condition_fields
```

Comment:
Nitpick: I find this repeated `serializer.fields[field_name].source` difficult to read, especially the second part `attrs[serializer.fields[field_name].source]`.

What do people think of splitting this statement over 2 lines?

```python
condition_sources = (serializer.fields[field_name].source for field_name in self.condition_fields)
condition_kwargs = {source: attrs[source] for source in condition_sources}
```